### PR TITLE
Check applied time on join

### PR DESCRIPTION
**Pull request by g4s8**
#956 - check applied time on join + test

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1115](https://github.com/zerocracy/farm/issues/1115) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1115](https://github.com/zerocracy/farm/issues/1115) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1582026698); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/zerocracy/tk/TkJoin.java [hunk 1]
```diff
@@ -85,6 +84,23 @@ public Response act(final RqRegex req) throws IOException {
...
+            people.appliedTime(author).toInstant(),
+            ZoneOffset.UTC
+        );
+        final long days = (long) new Policy().get("1.lag", 16);
+        if (when.isBefore(LocalDateTime.now(ZoneOffset.UTC).plusDays(days))) {
+            throw new RsForward(
+                new RsParFlash(
+                    new Par(
+                        "You can apply only one time in %d days, ",
+                        "you've applied %s"
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/tk/TkJoin.java]: @g4s8 one suggestion to make this message more user friendly. Let's also indicate how many days ago the user applied, e.g.:

>you've applied 2 days ago (2018-06-06)

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 2]
```diff
@@ -76,4 +79,31 @@ public void acceptRequestAndRedirectOnPost() throws Exception {
...
+        final Farm farm = new PropsFarm(new FkFarm());
+        final People people = new People(farm).bootstrap();
+        final String uid = "yegor256";
+        people.touch(uid);
+        people.apply(uid, new Date());
+        final RqWithUser req = new RqWithUser(
+            farm,
+            new RqFake("POST", "/join-post")
+        );
+        MatcherAssert.assertThat(
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @g4s8 what's with this assertion? Why are we making an iterable of blank string and checking it?

File: src/main/java/com/zerocracy/tk/TkJoin.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/zerocracy/tk/TkJoin.java]: @carlosmiranda right, updated message

File: src/test/java/com/zerocracy/tk/TkJoinTest.java (hunk 2 above)
Comment (g4s8) [reply src/test/java/com/zerocracy/tk/TkJoinTest.java]: @carlosmiranda sorry, forget to remove debug code

**Comment by g4s8** [comment]
@carlosmiranda fixed, please check

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1115](https://github.com/zerocracy/farm/issues/1115) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@g4s8 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>All commit messages have links to issues.

Commit with review fixes contains link to incorrect issue. Please confirm you'll try to avoid this mistake next time.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks

**Comment by g4s8** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@g4s8 Thank you/

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**